bundles/statistics/statsgrid/handler/StatisticsHandler.js

```diff
@@ -109,8 +109,12 @@ class StatisticsController extends StateHandlerBase {
         return updated;
     }
 
-    setActiveRegion (activeRegion) {
-        this.updateState({ activeRegion });
+    setActiveRegion (value) {
+        // toggle if already selected
+        const activeRegion = this.getState().activeRegion === value ? null : value;
+        // don't use normal updateState and notify with updated key to optimize map rendering
+        this.state = { ...this.getState(), activeRegion };
+        this.stateListeners.forEach(consumer => consumer(this.getState(), 'activeRegion'));
```

Comment:
This is a bit wild, but I don't have a better suggestion yet. Maybe we should add some other listener queue that RegionsetViewer could listen to instead, but lets let this simmer a bit and get back to it after 2.13.

Reply:
Suggested changes in https://github.com/okauppinen/oskari/pull/2